api/src/main/java/marquez/common/Utils.java

```diff
@@ -112,7 +117,46 @@ public static UUID toUuid(@NonNull final String uuidString) {
     return UUID.fromString(uuidString);
   }
 
-  public static Instant toInstant(@Nullable final String asIso) {
+  public static Instant toInstantOrNull(@Nullable final String asIso) {
     return (asIso == null) ? null : Instant.from(ISO_INSTANT.parse(asIso));
   }
+
+  public static SourceType sourceTypeFor(@NonNull final URI connectionUrl) {
+    return SourceType.of(toUrlWithNoJdbcPrefix(connectionUrl).getScheme());
+  }
+
+  public static URI urlWithNoCredentials(@NonNull final URI connectionUrl) {
```

Comment:
the name of the function doesn't match its behavior, I wouldn't expect it to modify jdbc specific urls.